sdk/purview/Azure.Analytics.Purview.Scanning/src/Generated/PurviewScan.cs

```diff
@@ -33,49 +33,48 @@ public partial class PurviewScanClient
+        internal PurviewScan(ClientDiagnostics clientDiagnostics, HttpPipeline pipeline, TokenCredential tokenCredential, Uri endpoint, string dataSourceName, string scanName, string apiVersion)
         {
-        }
-
-        /// <summary> Initializes a new instance of PurviewScanClient. </summary>
-        /// <param name="endpoint"> The scanning endpoint of your purview account. Example: https://{accountName}.scan.purview.azure.com. </param>
-        /// <param name="dataSourceName"> The String to use. </param>
-        /// <param name="scanName"> The String to use. </param>
-        /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
-        /// <param name="options"> The options for configuring the client. </param>
-        /// <exception cref="ArgumentNullException"> <paramref name="endpoint"/>, <paramref name="dataSourceName"/>, <paramref name="scanName"/> or <paramref name="credential"/> is null. </exception>
-        /// <exception cref="ArgumentException"> <paramref name="dataSourceName"/> or <paramref name="scanName"/> is an empty string, and was expected to be non-empty. </exception>
-        public PurviewScanClient(Uri endpoint, string dataSourceName, string scanName, TokenCredential credential, PurviewScanningServiceClientOptions options)
-        {
-            Argument.AssertNotNull(endpoint, nameof(endpoint));
-            Argument.AssertNotNullOrEmpty(dataSourceName, nameof(dataSourceName));
-            Argument.AssertNotNullOrEmpty(scanName, nameof(scanName));
-            Argument.AssertNotNull(credential, nameof(credential));
-            options ??= new PurviewScanningServiceClientOptions();
-
-            ClientDiagnostics = new ClientDiagnostics(options, true);
-            _tokenCredential = credential;
-            _pipeline = HttpPipelineBuilder.Build(options, Array.Empty<HttpPipelinePolicy>(), new HttpPipelinePolicy[] { new BearerTokenAuthenticationPolicy(_tokenCredential, AuthorizationScopes) }, new ResponseClassifier());
+            ClientDiagnostics = clientDiagnostics;
+            _pipeline = pipeline;
+            _tokenCredential = tokenCredential;
             _endpoint = endpoint;
             _dataSourceName = dataSourceName;
             _scanName = scanName;
-            _apiVersion = options.Version;
+            _apiVersion = apiVersion;
         }
 
         /// <summary> Get a filter. </summary>
         /// <param name="context"> The request context, which can override default behaviors on the request on a per-call basis. </param>
+        /// <example>
+        /// This sample shows how to call GetFilterAsync and parse the result.
+        /// <code><![CDATA[
+        /// var credential = new DefaultAzureCredential();
+        /// var endpoint = new Uri("<https://my-account-name.azure.com>");
+        /// var client = new PurviewScanningServiceClient(endpoint, credential).GetPurviewDataSourceClient("<dataSourceName>").GetPurviewScanClient("<scanName>");
```

Comment:
There are no methods named `GetPurviewDataSourceClient` or `GetPurviewScanClient`.  

Also, the data source client and scan clients are resource subclients, so the factory method should not have the `Client` suffix [according to the guidelines](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-subclient-factory-methods-suffix).

Reply:
>Would it be possible for you to confirm that the code compiles as part of your implementation?

For this, can we add these examples in a `.cs` file(maybe in `Generated/Examples` folder) and pull it into the docs from there(maybe use it as snippets)? If that's possible, then we can make sure that these examples compile.